jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -267,16 +267,13 @@ ecma_builtin_number_prototype_object_to_string (ecma_number_t this_arg_number, /
     'u', 'v', 'w', 'x', 'y', 'z'
   };
 
-  ecma_number_t arg_num;
-  ecma_value_t radix_num = ecma_get_number (arguments_list_p[0], &arg_num);
+  ecma_number_t radix;
 
-  if (!ecma_is_value_empty (radix_num))
+  if (ECMA_IS_VALUE_ERROR (ecma_op_to_integer (arguments_list_p[0], &radix)))
   {
-    return radix_num;
+    return ECMA_VALUE_ERROR;
   }
 
-  uint32_t radix = ecma_number_to_uint32 (arg_num);
-
   if (radix < 2 || radix > 36)
```

Comment:
After this point `radix` must be number in `[2, 36]`. So please cast it to `uint32` only once